### PR TITLE
feat: Add AnnData update, orchestration & CLI (PR 4)

### PR DESCRIPTION
**Pull request by Bento007**
This pull request adds a complete pipeline for annotating genetic perturbations in AnnData (`h5ad`) files with genomic locations and gene overlaps, using guidescan2 and bioframe. It introduces a new CLI command for this pipeline, implements robust helper functions for updating AnnData objects, and provides comprehensive tests to ensure correctness and schema compliance.

**Major features and changes:**

### New annotation pipeline and CLI integration

* Added a new public API function `annotate_perturbations_in_h5ad` in `annotate_guides.py` that orchestrates the full annotation pipeline: extracting perturbations, running guidescan2, annotating with bioframe, and updating the AnnData object.
* Introduced a new CLI command `annotate-perturbations` in `cli.py` to run the annotation pipeline from the command line, with options for specifying the guidescan2 index path.

### Helper functions for AnnData updates

* Implemented `_build_genomic_regions` and `_build_target_features` helpers to convert raw annotation data into schema-compliant formats (1-based, fully-closed coordinates and Ensembl IDs without version numbers).
* Added `update_h5ad_with_guide_annotations`, which updates only the relevant fields in `adata.uns['genetic_perturbations']`, with logic to skip guides with no gene overlaps and warn about missing guides.

### Comprehensive testing

* Added extensive unit and integration tests for the new annotation helpers and pipeline in `test_annotate_guides.py`, including edge cases for chromosome formatting, Ensembl ID version stripping, missing or extra guides, and error handling when guidescan2 is not installed.

These changes make it easy to annotate genetic perturbations in h5ad files via both API and CLI, while ensuring compliance with schema 7.1.0 and robust error handling throughout.

Depends on #1510 . Part of breaking down #1495 into smaller, reviewable PRs.

# Testing
1. Download the [Test dataset](https://www.dropbox.com/scl/fi/w4eop1ve910eilkf6ckrv/small_perturbation_dataset.h5ad?rlkey=nkb8esih0v8cqvrgtgpzoy4mi&dl=0) 
2. Install cellxgene-schema from the branch.
3. Run command: `cellxgene-schema annotate-perturbations`

### CODE REVIEW

File: cellxgene_schema_cli/cellxgene_schema/annotate_guides.py [hunk 1]
```diff
@@ -606,3 +607,172 @@ def annotate_guides_from_guidescan_csv(
...
+
+    genetic_perturbations = adata.uns["genetic_perturbations"]
+
+    # Filter annotations to only guides that exist in h5ad
+    h5ad_guide_ids = set(genetic_perturbations.keys())
+    extra_guides = set(annotations_df["id"].unique()) - h5ad_guide_ids
+
+    if extra_guides:
+        logger.warning(
+            f"Skipping {len(extra_guides)} guide(s) from annotations that are not in h5ad: " f"{sorted(extra_guides)}"
```
[the remark this answers is not in the document]
Comment (kliu-czi) [reply cellxgene_schema_cli/cellxgene_schema/annotate_guides.py]: I think this suggestion holds value

**Review by kliu-czi — APPROVED** [review]
